server/handler/leaderboard_config.go

```diff
@@ -0,0 +1,236 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/samouraiworld/topofgnomes/server/models"
+	"gorm.io/gorm"
+	clerk "github.com/clerk/clerk-sdk-go/v2"
+)
+
+// HandleGetLeaderboardConfigs returns all leaderboard configs for a given userId
+func HandleGetLeaderboardConfigs(db *gorm.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		claims, ok := clerk.SessionClaimsFromContext(r.Context())
+		if !ok || strings.TrimSpace(claims.Subject) == "" {
+			w.WriteHeader(http.StatusUnauthorized)
+			_ = json.NewEncoder(w).Encode(map[string]string{"error": "unauthorized"})
+			return
+		}
+		userID := claims.Subject
+
+		var configs []models.LeaderboardConfig
+		if err := db.Where("user_id = ?", userID).Order("id desc").Find(&configs).Error; err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			_ = json.NewEncoder(w).Encode(map[string]string{"error": err.Error()})
+			return
+		}
+		_ = json.NewEncoder(w).Encode(configs)
+	}
+}
+
+// HandleCreateLeaderboardConfig creates a new leaderboard config
+func HandleCreateLeaderboardConfig(db *gorm.DB) http.HandlerFunc {
+	type body struct {
+		ID                   uint   `json:"id"`
+		Platform             string `json:"platform"`
+		WebhookURL           string `json:"webhookUrl"`
+		SelectedRepositories string `json:"selectedRepositories"`
+		Timeline             string `json:"timeline"`
+		Frequency            string `json:"frequency"`
+		AnchorAt             string `json:"anchorAt"` // RFC3339
+		Timezone             string `json:"timezone"`
+		Active               *bool  `json:"active"`
+	}
```

Comment:
I think instead of using a different body type on Update and Create you can use directly the `models.LeaderboardConfig` This will allow you to just db.Save() the body. also you could have a simple function that receives this model.LeaderboardConfig and make all the checks you're doing now on both functions ?

wdyt.

This means that when updating you need to send the whole struct but don't see a problem with that :)